src/firefly/java/edu/caltech/ipac/firefly/util/ImageSetConverter.java

```diff
@@ -0,0 +1,209 @@
+                }
+            }
+
+        }
+        return params;
+    }
+
+    /**
+     * This method process each row data in the master-table.  Each row makes one JSONObject. All rows make
+     * an JSONArray of the size = number of rows
+     * @param row
+     * @param dataTypes
+     * @return
+     */
+    private JSONObject dataObjectToJSONObject(DataObject row,  DataType[] dataTypes){
+        JSONObject jsonObject = new JSONObject();
+
+
+        for (int i=0; i<dataTypes.length; i++){
+            Class cls = dataTypes[i].getDataType();
+            Object obj = row.getDataElement(dataTypes[i]);
+            if (cls==null){
+              jsonObject.put(dataTypes[i].getKeyName(),  obj);
+              continue;
+            }
+
+            switch (cls.getSimpleName().toLowerCase()){
+                case "string":
+                    String val = obj==null? null:((String) obj).trim();
+
+                    if (dataTypes[i].getKeyName().equalsIgnoreCase("missionLabel (project)")){
+                        jsonObject.put("project",  val);
+                    }
+                    else if ( dataTypes[i].getKeyName().equalsIgnoreCase("dataProductLabel") ){
+                        jsonObject.put("subProject",  val);
+                     }
+                    else if ( dataTypes[i].getKeyName().equalsIgnoreCase("wavebandDesc") ){
+                        jsonObject.put("title",  val);
+                    }
+                    else if ( dataTypes[i].getKeyName().equalsIgnoreCase("name") ){
+                        jsonObject.put("JSON DM-12001",  val);
+                    }
+                    else {
+                        jsonObject.put(dataTypes[i].getKeyName(),  val);
+                    }
+                    break;
+                case "float":
+                    float fval = obj==null? 0.1f:  ((Float)obj).floatValue();
+                    jsonObject.put(dataTypes[i].getKeyName(), fval);
+                    break;
+                case "integer":
+                    int ival = obj==null? 0:  ((Integer)obj).intValue();
+                    jsonObject.put(dataTypes[i].getKeyName(), ival);
+                    break;
+                case "double":
+                    double dval = obj==null? 0.1:  ((Double)obj).doubleValue();
+                    jsonObject.put(dataTypes[i].getKeyName(), dval );
+                    break;
+
+                default:
+                  jsonObject.put(dataTypes[i].getKeyName(),  obj);
+                  break;
+           }
+
+
+        }
+
+        HashMap<String, String> params = getMappedPlotRequestParam(row,dataTypes);
+        String[] keys = params.keySet().toArray(new String[0]);
+        JSONObject plotReqParams= new JSONObject();
+        plotReqParams.put("type", "SERVICE");
+        for(int i=0; i<params.size(); i++){
+            plotReqParams.put(keys[i], params.get(keys[i]));
+        }
+        jsonObject.put( "plotRequestParams",plotReqParams );
```

Comment:
Same comment as above.

Reply:
@ejoliet This is what you suggested at the first place: We hard coded the mapping now and later we will get them from the API.

Reply:
yes, is fine for now. But what i meant for API, is not related to the mapping itself, it needs to be 'hardcoded' outside because if API change (metadata columns), your code should be the same, the mapping will change. It could be read from props, but as i said, is fine for now. When we get to the point of getting it from API, we can move the mapping out.